x-pack/plugins/maps/public/classes/layers/vector_layer/vector_layer.tsx

```diff
@@ -332,29 +332,33 @@ export class VectorLayer extends AbstractLayer implements IVectorLayer {
     onLoadError,
     registerCancelCallback,
     dataFilters,
+    isForceRefresh,
   }: { join: InnerJoin } & DataRequestContext): Promise<JoinState> {
     const joinSource = join.getRightJoinSource();
     const sourceDataId = join.getSourceDataRequestId();
     const requestToken = Symbol(`layer-join-refresh:${this.getId()} - ${sourceDataId}`);
-    const searchFilters: VectorJoinSourceRequestMeta = {
-      ...dataFilters,
-      fieldNames: joinSource.getFieldNames(),
-      sourceQuery: joinSource.getWhereQuery(),
-      applyGlobalQuery: joinSource.getApplyGlobalQuery(),
-      applyGlobalTime: joinSource.getApplyGlobalTime(),
-      sourceMeta: joinSource.getSyncMeta(),
-    };
+
+    const joinRequestMeta = buildVectorRequestMeta(
```

Comment:
buildVectorRequestMeta returns VectorSourceRequestMeta instead of VectorJoinSourceRequestMeta. Is VectorJoinSourceRequestMeta still needed? Should you cast to VectorJoinSourceRequestMeta?. 

VectorJoinSourceRequestMeta does not very different from VectorSourceRequestMeta, only omitting geogridPrecision key.

Reply:
I don't mind the explicit type. It helps readability and documents internals. Will do the cast.